src/index.ts

```diff
@@ -10,71 +10,180 @@
+	started_at: string | null;
+	completed_at: string | null;
+	status: string;
+	input: {
+		[key: string]: any;
+	};
+	output: any | null;
+	error: string | null;
+	logs: string | null;
+	metrics: {
+		[key: string]: any;
+	};
+}
+
+const createS3Client = (env: Env) => new S3Client({
+	region: 'auto',
+	endpoint: `https://${env.CLOUDFLARE_ACCOUNT_ID}.r2.cloudflarestorage.com`,
+	credentials: {
+		accessKeyId: env.CLOUDFLARE_R2_ACCESS_KEY_ID,
+		secretAccessKey: env.CLOUDFLARE_R2_SECRET_ACCESS_KEY,
+	},
+});
+
+const handleCDNRequest = async (request: Request, env: Env, s3Client: S3Client) => {
+	const url = new URL(request.url);
+	const key = url.pathname.slice(5);
+	console.log('Key:', key);
+
+	try {
+		let cache = await env.BlogAssets.get<CachedSignedUrl>(key, `json`);
+		let signedUrl: string;
+
+		if (!cache || Date.now() > cache.refreshTime) {
+			const command = new GetObjectCommand({ Bucket: env.R2_BUCKET_NAME, Key: key });
+			signedUrl = await getSignedUrl(s3Client, command, { expiresIn: 3600 });
+			await env.BlogAssets.put(key, JSON.stringify({
+				url: signedUrl,
+				refreshTime: Date.now() + 55 * 60 * 1000,
+			}));
+		} else {
+			console.log(`Used cached signed URL for ${key}`);
+			signedUrl = cache.url;
+		}
+
+		const response = await fetch(signedUrl);
+		return new Response(response.body as BodyInit, {
+			status: response.status,
+			headers: response.headers,
+		});
+	} catch (error) {
+		console.error('Error generating signed URL:', error);
+		return new Response('Failed to generate signed URL', { status: 500 });
+	}
+};
+
+const generateImagePrompt = async (title: string, env: Env) => {
+	const anthropic = new Anthropic({ apiKey: env.ANTHROPIC_API_KEY });
+	const msg = await anthropic.messages.create({
+		model: "claude-3-5-sonnet-20240620",
+		max_tokens: 1000,
+		temperature: 0,
+		system: "reply only with the generated prompt and nothing else",
+		messages: [{
+			"role": "user",
+			"content": [{
+				"type": "text",
+				"text": `this is the prompt I have for the attached image:\n\nWatercolor style image on a textured white paper background. In the center, elegant hand-lettered text reads 'POSHMARK ALGORITHM REVEALED' in a deep purple color with a slight watercolor bleed effect. Surrounding the text, soft watercolor illustrations represent key aspects of the algorithm: a magnifying glass (for search), a clock face (for timing of shares), a golden trophy (for Poshmark Ambassador status), and a stylized feed icon (for feed visibility). Use a muted color palette with purple, teal, gold, and soft pink tones. The watercolor elements should have gentle color gradients and subtle bleeding effects, with some areas of the white paper showing through. Add a few splatter effects in the background for texture. In the bottom right corner, a small graph with a rising trend line painted in a loose, artistic style.\n\ncan you generate a similar prompt with creative materials relating to the blog post titled "${title}"?`
```

Comment:
Image isn't actually attached here. It works okay so far, but attached image should help improve prompt generation. Right now, the problem is in creating base64 image, [which is needed by Claude Vision model](https://docs.anthropic.com/en/docs/build-with-claude/vision), and creating it within the worker script results in memory issues.